### ISSUE TITLE
Linting error for fields with spaces in the name

### ISSUE THREAD

**Issue by jrod6810** [sqlfluff/sqlfluff#3119]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

There are certain tables in our database that have field names with spaces in them (e.g. `field name`). In order to properly query them, I wrap the field name in double quotes (e.g. `"field name"`). When running the sqlfluff linter, I get the error `L057: Do not use special characters in identifiers`. These field names are not something we will be able to change so I have typically added `--noqa: disable=L057` to my merge requests to disable this error from appearing. Ideally, these types of cases are not caught by the linter moving forward.

CC @PLBMR

### Expected Behaviour

No L057 error (or any other linting error) when I list a field name that has spaces wrapped in double-quotes (e.g. `"field name"`).

### Observed Behaviour

I observed an error when my SQL query referenced fields that had spaces in the name and were wrapped in double-quotes (e.g. `"field name"`).

### How to reproduce

```
select
    "field name 1" as field_name_1,
    "field name 2" as field_name_2
from table_name;
```

### Dialect

redshift

### Version

sqlfluff, version 0.11.1

### Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = redshift
templater = jinja
rules = None
exclude_rules = L009,L011,L019,L029,L031,L034
recurse = 0
output_line_length = 160
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 160
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = consistent

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Why not just turn off L057 if it’s not useful for you in `exclude_rules` like you have done for other rules that are not useful for your use of SQL?

**Comment by PLBMR** [comment]
If @jrod6810 were to resolve this flagged code, It would break his SQL query. I don't think our rules should be prescribing changes that break the underlying SQL.

Maybe we can make an exception for this rule for the Redshift dialect?

**Comment by tunetheweb** [comment]
Actually looking at the rule there is an `allow_space_in_identifier` config to rule L057. So you can just set that to True - in a similar way that you’ve configured L047 in your config above. This was added on SQLFluff 10.0.0.

> If @jrod6810 were to resolve this flagged code, It would break his SQL query. I don't think our rules should be prescribing changes that would break the underlying SQL query.

I completely disagree. If this rule auto fixed the code and broke the SQL that would be a problem I agree. But it doesn’t - precisely because of this risk.

A linter points out bad practices. And this is without doubt a bad practice. It’s up to you, the user, to decide what to do with that information and/or configure the linter to not flag items you do not wish flagged.

**Comment by PLBMR** [comment]
That makes sense to me. I was under the impression that the `fix` command would auto-fix this issue; since it doesn't, my concerns are alleviated.

@jrod6810 can you add this to your `.sqlfluff` file:

 ```
[sqlfluff:rules:L057]
allow_space_in_identifier = True
```

**Comment by tunetheweb** [comment]
Yeah a large number of our rules are auto fixable but not all of them. Some just can’t.

And we take issues that corrupt SQL pretty seriously and jump on those to resolve them. We have none of those currently: https://github.com/sqlfluff/sqlfluff/issues?q=is%3Aopen+is%3Aissue+label%3A%22fix+corrupts+sql%22 (thanks in most part to @barrywhart ‘s hard word!).

Cool so think this can be closed. Feel free to reopen if any more questions.
